templates/compose/supabase.yaml

```diff
@@ -597,6 +604,90 @@ services:
       # Use named volume to persist pgsodium decryption key between restarts
       - supabase-db-config:/etc/postgresql-custom
 
+  supabase-supavisor-setup:
+    image: curlimages/curl:8.8.0
+    restart: "no"
+    depends_on:
+      supabase-supavisor:
+        condition: service_healthy
+    environment:
+      - SETUP_SERVICE_SUPABASEANON_KEY=${SERVICE_SUPABASEANON_KEY}
+      - SETUP_POSTGRES_HOST=${POSTGRES_HOST:-supabase-db}
+      - SETUP_POSTGRES_PORT=${POSTGRES_PORT:-5432}
+      - SETUP_POSTGRES_DB=${POSTGRES_DB:-postgres}
+      - SETUP_SERVICE_PASSWORD_POSTGRES=${SERVICE_PASSWORD_POSTGRES}
+    entrypoint: [ "sh", "-c" ]
+    command: >
+      #!/bin/sh
+
+      while ! curl -sSfL --head -o /dev/null -H "Authorization: Bearer $SETUP_SERVICE_SUPABASEANON_KEY" http://supabase-supavisor:4000/api/health; do
+        sleep 2;
+      done;
+      
+      curl -X PUT 'http://supabase-supavisor:4000/api/tenants/dev_tenant' \
+      --header 'Accept: */*' \
+      --header 'User-Agent: Thunder Client (https://www.thunderclient.com)' \
+      --header 'Authorization: Bearer $SETUP_SERVICE_SUPABASEANON_KEY' \
+      --header 'Content-Type: application/json' \
+      --data-raw '{
+        "tenant": {
+          "db_host": "${SETUP_POSTGRES_HOST}",
+          "db_port": ${SETUP_POSTGRES_PORT},
+          "db_database": "${SETUP_POSTGRES_DB}",
+          "ip_version": "auto",
+          "enforce_ssl": false,
+          "require_user": false,
+          "auth_query": "SELECT rolname, rolpassword FROM pg_authid WHERE rolname=$1;",
+          "users": [
+            {
+              "db_user": "postgres",
+              "db_password": "${SETUP_SERVICE_PASSWORD_POSTGRES}",
+              "pool_size": 20,
+              "mode_type": "transaction",
+              "is_manager": true
+            }
+          ]
+        }
+      }'
+      exit 0
```

Comment:
```yaml
entrypoint: sh
command: /docker-entrypoint-initdb.d/supavisor.sh
volumes:
  -
    type: bind
    source: ./volumes/supavisor.sh
    target: /docker-entrypoint-initdb.d/supavisor.sh
    content: |
        while ! curl -v -H "Authorization: Bearer ${SETUP_SERVICE_SUPABASEANON_KEY}" http://supabase-supavisor:4000/api/health; 
        do sleep 2;
        done;
        
        curl  -X PUT -v \
          'http://supabase-supavisor:4000/api/tenants/dev_tenant' \
          --header 'Accept: */*' \
          --header 'User-Agent: Thunder Client (https://www.thunderclient.com)' \
          --header "Authorization: Bearer ${SETUP_SERVICE_SUPABASEANON_KEY}" \
          --header 'Content-Type: application/json' \
          --data '
        {
          "tenant": {
            "db_host": "'"${SETUP_POSTGRES_HOST}"'",
            "db_port": '$SETUP_POSTGRES_PORT',
            "db_database": "postgres",
            "ip_version": "auto",
            "enforce_ssl": false,
            "require_user": false,
            "auth_query": "SELECT rolname, rolpassword FROM pg_authid WHERE rolname=$1;",
            "users": [
              {
                "db_user": "postgres",
                "db_password": "'"$SETUP_SERVICE_PASSWORD_POSTGRES"'",
                "pool_size": 20,
                "mode_type": "transaction",
                "is_manager": true
              },
            ]
          }
        }'
        
        exit 0
``` 

fixes the formatting issues in the compose and vars in single quotes
also makes it much easier to add extra users, since you prob dont want to be using postgres user account you can just edit storage file